<launch>
    <arg name="robot1_active" default="false"/>
    <arg name="robot2_active" default="false"/>
    <arg name="rival1_active" default="true"/>
    <arg name="rival2_active" default="false"/>

    <group ns="survive1" if="$(arg robot1_active)">
        <node name="vive_world" pkg="vive" type="vive_world" output="screen">
            <rosparam file="$(find vive)/param/vive_localization.yaml" command="load" />
            <param name ="tracker" type="string" value="trackerA"/> <!--the tracker used for publishing velocity-->
            <param name ="survive_prefix" type="string" value="survive1_"/>
            <param name ="vel_topic_name" type="string" value="/robot1/tracker_vel"/>
            <param name ="max_distance_bt_maps" type="double" value="0.1"/>
        </node>
    </group>

    <group ns="survive2" if="$(arg robot2_active)">
        <node name="vive_world" pkg="vive" type="vive_world" output="screen">
            <rosparam file="$(find vive)/param/vive_localization.yaml" command="load" />
            <param name ="tracker" type="string" value="trackerA"/> <!--the tracker used for publishing velocity-->
            <param name ="survive_prefix" type="string" value="survive2_"/>
            <param name ="vel_topic_name" type="string" value="/robot2/tracker_vel"/>
            <param name ="max_distance_bt_maps" type="double" value="0.1"/>
        </node>
    </group>
    
    <group ns="survive3" if="$(arg rival1_active)">
        <node name="vive_world" pkg="vive" type="vive_world" output="screen">
            <rosparam file="$(find vive)/param/vive_localization.yaml" command="load" />
            <param name ="tracker" type="string" value="trackerC"/> <!--the tracker used for publishing velocity-->
            <param name ="survive_prefix" type="string" value="survive3_"/>
            <param name ="vel_topic_name" type="string" value="/rival1/tracker_vel"/>
            <param name ="max_distance_bt_maps" type="double" value="0.1"/>
        </node>
    </group>

    <group ns="survive4" if="$(arg rival2_active)">
        <node name="vive_world" pkg="vive" type="vive_world" output="screen">
            <rosparam file="$(find vive)/param/vive_localization.yaml" command="load" />
            <param name ="tracker" type="string" value="trackerA"/> <!--the tracker used for publishing velocity-->
            <param name ="survive_prefix" type="string" value="survive4_"/>
            <param name ="vel_topic_name" type="string" value="/rival2/tracker_vel"/>
            <param name ="max_distance_bt_maps" type="double" value="0.1"/>
        </node>
    </group>
    <!-- <node name="vive_trackerpose" pkg="vive" type="vive_trackerpose" output="screen">
        <param name ="freq" type="int" value="21"/>
        <param name ="unit" type="int" value="1"/>
    </node> -->

</launch>